clients/js-legacy/src/actions/amountToUiAmount.ts

```diff
@@ -144,61 +189,64 @@ export async function amountToUiAmountForMintWithoutSimulation(
 ): Promise<string> {
     const accountInfo = await connection.getAccountInfo(mint);
     const programId = accountInfo?.owner;
+
     if (programId !== TOKEN_PROGRAM_ID && programId !== TOKEN_2022_PROGRAM_ID) {
         throw new Error('Invalid program ID');
     }
 
     const mintInfo = unpackMint(mint, accountInfo, programId);
+    const timestamp = await getSysvarClockTimestamp(connection);
```

Comment:
This is introducing a network call to get the timestamp for all mints -- even if there's some copy-pasta, it's better to have this line in both branches (mint-with-interest-bearing and mint-with-scaled-ui-amount) to avoid always fetching the clock.

Or we can add an `if (!interestBearingMintConfigState && !scaledUiAmountConfig)` branch earlier, similar to how the code was before